src/main/kotlin/frc/robot/robotstate/RobotCommands.kt

```diff
@@ -0,0 +1,53 @@
+package frc.robot.robotstate
+
+import edu.wpi.first.units.measure.Distance
+import edu.wpi.first.wpilibj2.command.Commands
+import edu.wpi.first.wpilibj2.command.ConditionalCommand
+import frc.robot.drive
+import frc.robot.flywheel
+import frc.robot.hood
+import frc.robot.hopper
+import frc.robot.lib.extensions.distanceFromPoint
+import frc.robot.lib.extensions.get
+import frc.robot.lib.extensions.m
+import frc.robot.lib.extensions.rad
+import frc.robot.lib.extensions.rotationFromPoint
+import frc.robot.lib.extensions.rps
+import frc.robot.lib.getPose2d
+import frc.robot.roller
+import frc.robot.subsystems.drive.alignToPose
+import frc.robot.turret
+
+val robotDistanceFromBasket
+    get() = drive.pose.distanceFromPoint(BASKET_LOCATION.translation)
+val turretRotationToBasket
+    get() = drive.pose.rotationFromPoint(BASKET_LOCATION.translation)
+
+fun poseToMoveTo(distance: Distance) =
+    getPose2d(
+        drive.pose.translation / (robotDistanceFromBasket[m] / distance[m])
+    )
+
+fun driveToShootingPoint() =
+    ConditionalCommand(
+        drive.defer { alignToPose(poseToMoveTo(MIN_DISTANCE_FROM_BASKET)) },
+        drive.defer { alignToPose(poseToMoveTo(MAX_DISTANCE_FROM_BASKET)) },
+        isOuterDeadZone
+    ).withName("RobotCommands/driveToShootingPoint")
```

Comment:
This could be more concise:
```suggestion
fun DriveToShootingPoint() = drive.defer {
    val distance = if (isOuterDeadZone) minDistanceFromBasket else MaxDistanceFromBasket
    val shootingPoint = poseToMoveTo(MinDistanceFromBasket)
    alignToPose(shootingPoint)
}.withName("Drive/Drive to shooting point")
```